File: haskell/NTL_1_A.hs
primes = 2 : 3 : 5 : primes'
  where
    isPrime (p:ps) n = p*p > n || n `rem` p /= 0 && isPrime ps n
    primes' = 7 : filter (isPrime primes') (scanl (+) 11 $ cycle [2,4,2,4,6,2,6,4])

decomp 1 _ = []
decomp n (p:ps) = 
    if n < p*p
    then [n]
    else if r == 0
         then p:(decomp n' (p:ps))
         else decomp n ps
    where
        r = n `mod` p
        n' = n `div` p

ans n = 
     let d = decomp n primes
     in
         (show n) ++ ": " ++ (unwords $ map show d)

main = do
    n <- getLine
    let i = read n :: Int
        o = ans i
    putStrLn o

-- time limit exceeded

File: python/NTL_1_A.py
a = int(input())

ans = []
val = a

while True:
    if val % 2 == 0:
        val //= 2
        ans.append(2)
    else:
        break
    
div = 3
while True:
    if val == 1:
        break
    elif val < div * div:
        break
    elif val % div == 0:
        val //= div
        ans.append(div)
    else:
        div += 2

if val != 1:
    ans.append(val)
    
print(f"{a}:", " ".join(map(str, ans)))
